apps/judicial-system/backend/src/app/modules/subpoena/subpoena.service.ts

```diff
@@ -0,0 +1,140 @@
+import { Includeable, Sequelize } from 'sequelize'
+import { Transaction } from 'sequelize/types'
+
+import { forwardRef, Inject, Injectable } from '@nestjs/common'
+import { InjectConnection, InjectModel } from '@nestjs/sequelize'
+
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+
+import type { User } from '@island.is/judicial-system/types'
+
+import { Case } from '../case/models/case.model'
+import { Defendant } from '../defendant/models/defendant.model'
+import { PoliceService } from '../police'
+import { UpdateSubpoenaDto } from './dto/updateSubpoena.dto'
+import { DeliverResponse } from './models/deliver.response'
+import { Subpoena } from './models/subpoena.model'
+
+export const include: Includeable[] = [
+  { model: Case, as: 'case' },
+  { model: Defendant, as: 'defendant' },
+]
+@Injectable()
+export class SubpoenaService {
+  constructor(
+    @InjectConnection() private readonly sequelize: Sequelize,
+    @InjectModel(Subpoena) private readonly subpoenaModel: typeof Subpoena,
+    @InjectModel(Defendant) private readonly defendantModel: typeof Defendant,
+    @Inject(forwardRef(() => PoliceService))
+    private readonly policeService: PoliceService,
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+  ) {}
+
+  async createSubpoena(defendant: Defendant): Promise<Subpoena> {
+    return await this.subpoenaModel.create({
+      defendantId: defendant.id,
+      caseId: defendant.caseId,
+    })
+  }
+
+  async update(
+    subpoena: Subpoena,
+    update: UpdateSubpoenaDto,
+    transaction?: Transaction,
+  ): Promise<Subpoena> {
+    const {
+      defenderChoice,
+      defenderNationalId,
+      defenderEmail,
+      defenderPhoneNumber,
+      defenderName,
+    } = update
+
+    const [numberOfAffectedRows] = await this.subpoenaModel.update(update, {
+      where: { subpoenaId: subpoena.subpoenaId },
```

Comment:
Consider adding `caseId` to the where clause to prefent fraudulent subpoena updates. It may not be necessary if this is only used for internal updates.

Reply:
The update request that ends up is initialized from RLS, and they don't have our caseId logged for use on their side, so the only way we could use the caseId here would by by looking it up from the subpoenaId which would make it redundant or by having them retrieve or store it on their side. I'm definitely up for discussion about it if it's something we're worried about.